compiler/rustc_parse/src/parser/path.rs

```diff
@@ -134,7 +134,15 @@ impl<'a> Parser<'a> {
         maybe_whole!(self, NtPath, |path| {
             if style == PathStyle::Mod && path.segments.iter().any(|segment| segment.args.is_some())
             {
-                self.struct_span_err(path.span, "unexpected generic arguments in path").emit();
+                self.struct_span_err(
+                    path.segments
+                        .iter()
+                        .filter_map(|segment| segment.args.as_ref())
+                        .map(|arg| arg.span())
+                        .collect::<Vec<_>>(),
```

Comment:
can you build a test that has two unexpected generics within a single path?

This is only about module paths, right? so any generic is unexpected right now?

Reply:
Yes, they are already unexpected, the change is that the primary span will only point at the generic argument instead of the whole path. I noticed this in an error affecting macro input. I've been making these kind of changes piece-meal as I encounter them. I no longer expect this kind of behavior (pointing at more than needed) to be prevalent, which means that when we encounter them is in obscure places :)